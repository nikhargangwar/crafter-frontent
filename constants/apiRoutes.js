const userRoute = 'api/user/';

export default { userRoute };