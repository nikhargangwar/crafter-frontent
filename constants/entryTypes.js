const entryTypes = {
  userConfig :{
    FRONTEND:{
      port:'integer',
      numberOfReplicas : 'integer',
      name:'string',
    },
    BACKEND:{
      port:'integer',
      numberOfReplicas:'integer',
      name:'string',
    },
    DATABASE:{
      port:'integer',
      numberOfReplicas:'integer',
      dbUser:'string',
      dbPassword:'string',
      name:'string',
      schemaName :'string',
    }},
  dockerConfig:{
    username:'string',
    email:'e-mail',
    repositoryImageAddress : 'string',
    token:'string'},
};

export default entryTypes;
