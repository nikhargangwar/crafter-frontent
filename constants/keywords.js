const arrayOfWordsForLandingPage = [ ' containerized applications on Kubernetes with our intuitive K8s Designer tool.', ' Frontend - React.', ' Backend - Node.js.', ' Database - Postgres.' ];

export default arrayOfWordsForLandingPage;