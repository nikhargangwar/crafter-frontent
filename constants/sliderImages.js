import loignSliderImg1 from '../assets/undraw_code_thinking_re_gka2.svg';
import loginSliderImg2 from '../assets/undraw_dev_focus_re_6iwt.svg';
import loginSliderImg3 from '../assets/undraw_specs_re_546x.svg';
import loginSliderImg4 from '../assets/undraw_coding_re_iv62.svg';
import signupSliderImg1 from '../assets/undraw_programmer_re_owql.svg';
import signupSliderImg2 from '../assets/undraw_programming_re_kg9v.svg';
import signupSliderImg3 from '../assets/undraw_dev_productivity_re_fylf.svg';
import signupSliderImg4 from '../assets/undraw_in_the_office_re_jtgc.svg';


const loginImages = [
  {
    original: loignSliderImg1
  },
  {
    original: loginSliderImg2
  },
  {
    original: loginSliderImg3
  },
  {
    original: loginSliderImg4
  }
];

const signupImages = [
  {
    original: signupSliderImg1
  },
  {
    original: signupSliderImg2
  },
  {
    original: signupSliderImg3
  },
  {
    original: signupSliderImg4
  }
];

export default { loginImages, signupImages };