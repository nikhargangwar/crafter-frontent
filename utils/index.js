import CustomNode from './CustomNode';
import FormValidation from './FormValidation';
import makeRequest from './makeRequest';
import {camelize, getConfiguration, getSpaceSeparated, updateConfiguration} from './ServiceHandler';
export { CustomNode, FormValidation, makeRequest, camelize, getConfiguration, getSpaceSeparated, updateConfiguration };