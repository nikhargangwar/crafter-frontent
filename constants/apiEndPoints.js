import apiRoutes from './apiRoutes';

const GET_ALL_BLUEPRINTS = () => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: '/blueprints',
  method: 'GET',
});

const GET_BLUEPRINT_METADATA = (blueprintId) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}`,
  method: 'GET',
});

const GET_IMAGES =(name)=>({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/minio/images/${name}`,
  method: 'GET',
});

const GET_SERVICES_BY_BLUEPRINT_ID = (blueprintId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services`,
  method: 'GET',
});

const CREATE_BLUEPRINT = () => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: '/blueprints',
  method: 'POST',
});

const DELETE_BLUEPRINT = (blueprintId) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}`,
  method: 'DELETE',
});

const CREATE_NEW_SERVICE =(blueprintId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services`,
  method: 'POST'
});

const UPDATE_SERVICE_POSITION= (blueprintId, serviceId, version) =>({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}`,
  method: 'PUT',
});

const registerUser = (name, otp, email, password) => {
  return {
    baseURL: process.env.REACT_APP_API_GATEWAY_URL,
    url: apiRoutes.userRoute + 'register',
    method: 'post',
    data: {
      name,
      otp,
      email,
      password,
    }
  };
};

const loginUser = (email, password) => {
  return {
    baseURL: process.env.REACT_APP_API_GATEWAY_URL,
    url: apiRoutes.userRoute + 'login',
    method: 'post',
    data: {
      email,
      password,
    }
  };
};

const validateUser = (token) => {
  return {
    baseURL: process.env.REACT_APP_API_GATEWAY_URL,
    url: apiRoutes.userRoute + 'validate',
    method: 'get',
    headers: {
      authorization: token
    }
  };

};

const generateOtp = (email) => {
  return {
    baseURL: process.env.REACT_APP_API_GATEWAY_URL,
    url: apiRoutes.userRoute + 'generate/otp',
    method: 'post',
    data: {
      email,
    }
  };
};

const GET_SYSTEM_COMPONENTS = () => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: '/plugins/types/SystemComponent',
  method: 'GET',
});

const GET_DEPLOYMENT_STACK = () => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: '/plugins/types/DeploymentStack',
  method: 'GET',
});

const GET_PLUGIN_BY_ID = (pluginId) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/plugin/${pluginId}`,
  method: 'GET',
});


const CONNECT_SERVICES = (blueprintId,serviceId,version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}/connections`,
  method: 'POST',

});

const DISCONNECT_SERVICES = (blueprintId,serviceId,version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}/connections`,
  method: 'DELETE',
});

const GET_SERVICE_BY_ID = (blueprintId, serviceId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}`,
  method: 'GET',
});

const UPDATE_CONFIGURATIONS = (blueprintId, serviceId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}/configurations`,
  method: 'PUT',
});

const UPDATE_METADATA = (blueprintId, serviceId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}/metadata`,
  method: 'PUT',
});

const DELETE_SERVICE_BY_ID = (blueprintId, serviceId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}`,
  method: 'DELETE',
});

const UPDATE_BLUEPRINT_METADATA = (blueprintId) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}`,
  method: 'PUT',
});

const SEARCH_CONFIGURATIONS = (blueprintId, serviceId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/services/${serviceId}/configurations`,
  method: 'GET',
});

const GET_ALL_CLUSTERS = () => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: '/clusters',
  method: 'GET',
});

const GET_CLUSTER_BY_ID = (clusterId) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/clusters/${clusterId}`,
  method: 'GET',
});

const CREATE_CLUSTER = () => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: '/clusters',
  method: 'POST',
});

const GET_ALL_TASKS= () => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: '/tasks',
  method: 'GET',
  
});

const GET_TASKS_BY_ID = (id) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/tasks/${id}`,
  method: 'GET',
});

const DOWNLOAD_BOILERPLATE = (objectName) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/minio/boilerplate/${objectName}`,
  method: 'GET',
});

const DELETE_CLUSTER = (clusterId) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/clusters/${clusterId}`,
  method: 'DELETE',
});

const EXPORT_BLUEPRINT = (blueprintId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/blueprints/${blueprintId}/versions/${version}/exports`,
  method: 'POST',
});

const DEPLOY_CLUSTER = (clusterId,blueprintId, version) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/clusters/${clusterId}/blueprints/${blueprintId}/versions/${version}/deploy`,
  method: 'POST',
});

const TEST_CONNECTION = (clusterId) => ({
  baseURL: process.env.REACT_APP_API_GATEWAY_URL,
  url: `/clusters/${clusterId}/testConnection`,
  method: 'POST',
});

export default { 
  generateOtp,
  registerUser,
  loginUser,
  validateUser,
  GET_ALL_BLUEPRINTS,
  GET_BLUEPRINT_METADATA,
  CREATE_BLUEPRINT,
  DELETE_BLUEPRINT,
  GET_SYSTEM_COMPONENTS,
  GET_SERVICES_BY_BLUEPRINT_ID,
  GET_SERVICE_BY_ID,
  UPDATE_CONFIGURATIONS,
  UPDATE_METADATA,
  CREATE_NEW_SERVICE,
  DELETE_SERVICE_BY_ID,
  CONNECT_SERVICES,
  DISCONNECT_SERVICES,
  UPDATE_BLUEPRINT_METADATA,
  UPDATE_SERVICE_POSITION,
  SEARCH_CONFIGURATIONS,
  GET_ALL_CLUSTERS,
  GET_CLUSTER_BY_ID,
  CREATE_CLUSTER,
  GET_DEPLOYMENT_STACK,
  GET_ALL_TASKS,
  GET_TASKS_BY_ID,
  DOWNLOAD_BOILERPLATE,
  GET_IMAGES,
  GET_PLUGIN_BY_ID,
  DELETE_CLUSTER,
  EXPORT_BLUEPRINT,
  DEPLOY_CLUSTER,
  TEST_CONNECTION
};
