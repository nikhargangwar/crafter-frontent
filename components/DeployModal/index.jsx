import { Alert, Autocomplete, Box, Button, Modal, Portal, Snackbar, TextField, Typography } from '@mui/material';
import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { Close } from '@mui/icons-material';
import { modalStyle } from '../../constants/styles';
import { makeRequest } from '../../utils';
import { apiEndPoints } from '../../constants';
import { useNavigate } from 'react-router-dom';

function DeployModal({openModal, setOpenModal, blueprintId, cluster = null, version}) {
  const [activeStep, setActiveStep] = useState(cluster ? 1 : 0);
  const [clusters, setClusters] = useState([]);
  const [selectedCluster, setSelectedCluster] = useState({});
  const [credentials, setCredentials] = useState({});
  const [connectionSuccess, setConnectionSuccess] = useState(false);
  const [snackbar, setSnackbar] = useState({
    status: false,
    message: '',
    type: ''
  });
  const navigate = useNavigate();
  const token = localStorage.getItem('jwt_token');

  useEffect(() => {
    makeRequest.makeRequest(apiEndPoints.GET_ALL_CLUSTERS(), navigate, token)
      .then((response) => {
        setClusters(response);
      });
  }, []);

  const handleDeploy = () => {
    const request = {
      data: {
        ...credentials,
        clusterName: cluster || selectedCluster.clusterName,
      }};
    makeRequest.makeRequest(apiEndPoints.DEPLOY_CLUSTER(selectedCluster.id, blueprintId, version), null, token, request)
      .then(() => {
        setSnackbar({
          status: true,
          message: 'Deployment in progress',
          type: 'success'
        });
        setTimeout(() => {
          navigate(`/tasks/${blueprintId}`);
        }, 5000);
      })
      .catch(() => {
        setSnackbar({
          status: true,
          message: 'Deployment Failed',
          type: 'error'
        });
      });
  };

  const handleTestConnection = () => {
    const request = selectedCluster.name === 'local' ? null : {
      data: {
        ...credentials,
        clusterName: cluster || selectedCluster.clusterName,
      }
    };
    makeRequest.makeRequest(apiEndPoints.TEST_CONNECTION(selectedCluster.id), null, token, request)
      .then(() => {
        setSnackbar({
          status: true,
          message: 'Connection Successful',
          type: 'success'
        });
        setConnectionSuccess(true);
      })
      .catch(() => {
        setSnackbar({
          status: true,
          message: 'Connection Failed',
          type: 'error'
        });
      });
  };

  return (
    <>
      <Modal
        open={openModal}
        onClose={() => {
          setOpenModal(false);
          setActiveStep(0);
        }}
      >
        <Box sx={modalStyle}>
          <div className='flex justify-between items-center mb-5'>
            <Typography variant='h5' style={{ fontWeight:'bold' }}>{(activeStep) ? 'Enter Credentials' : 'Select Cluster' }</Typography>
            <Close onClick={() => {
              setOpenModal(false);
              setActiveStep(0);
            }}
            className='cursor-pointer'/>
          </div>
          {activeStep === 0 && <>
            <Autocomplete
              options={clusters}
              getOptionLabel={(option) => option.clusterName}
              className='my-5 rounded-md'
              onChange={(e, value) => setSelectedCluster(value)}
              renderInput={(params) =>
                <TextField {...params}
                  label='Select Cluster'
                  fullWidth
                  variant='outlined'
                  required
                />}
            />
            { selectedCluster?.clusterName === 'local' ? 
              <div className='flex justify-end gap-2 mt-4'>
                <Button
                  variant='contained'
                  className='mr-2'
                  sx={{bgcolor:'primary.main',color:'white',fontWeight:'bold', fontSize:'16px'}}
                  onClick={handleTestConnection}
                >
                Test Connection
                </Button>
                <Button
                  disabled={!connectionSuccess}
                  variant='contained'
                  className='mr-2'
                  sx={{bgcolor:'primary.main',color:'white',fontWeight:'bold', fontSize:'16px'}}
                  onClick={handleDeploy}
                >
                Deploy
                </Button>
              </div> :
              <div className='flex justify-end'>
                <Button
                  variant='contained'
                  className='mr-2'  
                  sx={{bgcolor:'primary.main',color:'white',fontWeight:'bold', fontSize:'16px'}}
                  onClick={() => {
                    setActiveStep(1);
                  }}
                >
              Next
                </Button>
              </div>
            }
          </>}
          {activeStep === 1 && <>
            <div className='flex flex-col gap-2 mt-2'>
              <TextField
                label='AWS Access Key Id'
                fullWidth
                variant='outlined'
                className='my-5 rounded-md'
                required
                onChange={(e) => setCredentials((prev) => ({ ...prev, awsAccessKeyId: e.target.value }))}
              />
              <TextField
                label='AWS Secret Access Key'
                fullWidth
                variant='outlined'
                className='my-5 rounded-md'
                required  
                type='password'
                onChange={(e) => setCredentials((prev) => ({ ...prev, awsSecretAccessKey: e.target.value }))}
              />
              <TextField
                label='AWS Session Token'
                fullWidth
                variant='outlined'
                className='my-5 rounded-md'
                required  
                type='password'
                onChange={(e) => setCredentials((prev) => ({ ...prev, awsSessionToken: e.target.value }))}
              />
            </div>
            <div className='flex justify-end gap-2 mt-4'>
              <Button
                disabled={Object.keys(credentials).length < 2}
                variant='contained'
                className='mr-2'
                sx={{bgcolor:'primary.main',color:'white',fontWeight:'bold', fontSize:'16px'}}
                onClick={handleTestConnection}
              >
              Test Connection
              </Button>
              <Button
                disabled={!connectionSuccess}
                variant='contained'
                className='mr-2'
                sx={{bgcolor:'primary.main',color:'white',fontWeight:'bold', fontSize:'16px'}}
                onClick={handleDeploy}
              >
              Deploy
              </Button>
            </div>
          </>}
        </Box>
      </Modal>
      <Portal>
        <Snackbar open={snackbar.status} autoHideDuration={6000} onClose={() => setSnackbar({ ...snackbar, status: false })}>
          <Alert onClose={() => setSnackbar({ ...snackbar, status: false })} severity={snackbar.type} sx={{ width: '100%' }}>
            {snackbar.message}
          </Alert>
        </Snackbar>
      </Portal>
    </>
  );
}

export default DeployModal;

DeployModal.propTypes = {
  openModal: PropTypes.bool.isRequired,
  setOpenModal: PropTypes.func.isRequired,
  blueprintId: PropTypes.string,
  cluster: PropTypes.string,
  version: PropTypes.string,
  selectedClusterId: PropTypes.string,
};

