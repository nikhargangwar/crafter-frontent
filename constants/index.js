import apiEndPoints from './apiEndPoints';
import apiRoutes from './apiRoutes';
import entryTypes from './entryTypes';
import arrayOfWordsForLandingPage from './keywords';
import sliderImages from './sliderImages';

export { apiEndPoints, apiRoutes, entryTypes, arrayOfWordsForLandingPage, sliderImages };
