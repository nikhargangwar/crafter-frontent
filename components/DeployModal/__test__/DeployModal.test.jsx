import React from 'react';
import { fireEvent, render, screen} from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import DeployModal from '..';
import { makeRequest } from '../../../utils';
import { apiEndPoints } from '../../../constants';

describe('DeployModal component', () => {
  it('should render correctly the first window', () => {
    const { asFragment } = render(<Router>
      <DeployModal 
        openModal={true}
        setOpenModal={() => {}}
        blueprintId='1'
        cluster='test'
        version='1'
      />
    </Router>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should display all clusters', async () => {
    jest.spyOn(makeRequest, 'makeRequest').mockResolvedValue([
      {
        id: 1,
        clusterName: 'test name 1',
      },
      {
        id: 2,
        clusterName: 'test name 2',
      }
    ]);
    render(<Router>
      <DeployModal
        openModal={true}
        setOpenModal={() => {}}
        blueprintId='1'
        version='1'
      />
    </Router>);
    const clusterDropdown = screen.getByTestId('ArrowDropDownIcon');
    expect(clusterDropdown).toBeTruthy();
    await fireEvent.click(clusterDropdown);
    const clusters = await screen.findAllByText(/test name/i);
    expect(clusters).toHaveLength(2);
  });
});