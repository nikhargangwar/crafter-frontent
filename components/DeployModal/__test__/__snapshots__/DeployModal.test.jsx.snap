// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeployModal component should render correctly the first window 1`] = `<DocumentFragment />`;
